dali/pipeline/operator/operator.h

```diff
@@ -166,6 +168,23 @@ class DLL_PUBLIC OperatorBase {
 
   DISABLE_COPY_MOVE_ASSIGN(OperatorBase);
 
+
+  template<typename T>
+  T GetDiagnostic(const std::string &name) const {
+    return *any_cast<T *>(diagnostics_.at(name));
+  }
+
+  template<typename T>
+  void RegisterDiagnostic(std::string name, T *val) {
+    using namespace std;  // NOLINT
+    static_assert(is_arithmetic_or_half<remove_reference_t<T>>::value || is_same<T, string>::value,
```

Comment:
I'd say that the limitation should be to POD types. `string` is dangerous - it's a complex type with internal dynamic storage and requirements for consistency - unguarded parallel access may result in fatal race conditions.

Reply:
removed string, added enum